app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -244,4 +258,21 @@ class ManageObligationsController @Inject()(val checkSessionTimeout: SessionTime
             Future.successful(Redirect(controllers.incomeSources.manage.routes.ManageIncomeSourceController.show(true)))
         }
   }
+
+  private def authenticatedAction(isAgent: Boolean
```

Comment:
I am not sure why we need at this level; this a good candidate to be moved to the base controller level or we can design a new trait?